docs/architecture/adr-003-module-sub-accounts.md

```diff
@@ -0,0 +1,112 @@
+# ADR 3: Module Sub-Accounts
+
+## Changelog
+
+## Context
+
+Currently `ModuleAccount`s must be declared upon Supply Keeper initialization. In addition to this they don't allow for separation of fungible coins within an account.
+
+We want to support the ability to define and manage sub-module-accounts.
+
+## Decision
+
+We will modify the existing `ModuleAccount` interface to support a heirarchical account structure.
+The `ModuleAccount`s defined upon initialization of Supply Keeper are the roots of family trees.
+Each `ModuleAccount` in a family tree may have zero or more children.
+A `ModuleAccount` with one or more children is considered a parent to each child.
+All `ModuleAccount`s have exactly one parent, unless they are the root of their family tree.
+
+`ModuleAccount` permissions will be renamed to `Attribute`.
+Each child's attributes must be a subset of their parent's attributes.
+There is no limit on the number of children a `ModuleAccount` can have.
+No `ModuleAccount`s can be removed from a family tree.
+A `ModuleAccount` name is the path of the `ModuleAccount` names used to reach the child.
+It starts with the root `ModuleAccount` name and is separated by a colon for each parent that follows until the child is reached.
+
+Example name: `root:parent:child`
+
+We will add a `TrackBalance` function which recursively updates the passive tracking of parent balances.
+A `ModuleAccount` address is the hash of its name.
+A `ModuleAccount` has no pubkeys.
+The function `AddChildToModuleAccount` will be added to Supply Keeper, 
+It will validate that the granted attributes are a subset of the parent and then register the child's name with the Supply Keeper.
+
+### Implementation Changes
+
+Modify `ModuleAccount` interface in `x/supply`:
+
+```go
+type ModuleAccount interface {
```

Comment:
Need to have some validation to make sure ModuleAccount name doesn't contain delimiter.
i.e. No name should contain ":"